### DISCUSSION TITLE
%d format: a number is required, not str

### DISCUSSION THREAD

**Discussion by XPPA0215** [PyMySQL/PyMySQL#1151] [Q&A] [answered]
Code:

```
...
sql = '''SELECT aa, bb, refresh_time FROM table_name_%03d WHERE id = %s ORDER BY refresh_time DESC'''
...
            cursor.execute(sql, ((id / 10) % 1000), id)
            data = cursor.fetchall()
```

**Additional context**
I konw,PyMySQL convert all objects including integer to formatted and quoted string.
But when my table name is table_name_003.I expected it to be table_name_003 but I got table_name_'003'
I have also tried to make mysql not add automatically ''.But I failed.
Do you have any good suggestions, please

**Comment by methane** [comment]
Do not use placeholders for regular string interpolation. Use Python features instead.

**Reply by XPPA0215** [reply]
Thanks for reply，Your suggestion is to use f"...%(name)s..."？
I tried, too.But here's what I got: select * from 'table_name'.
This is wrong

**Reply by methane** [reply]
```py
...
table_name = "table_name_%03d" % ((id / 10) % 1000)
sql = f'''SELECT aa, bb, refresh_time FROM {table_name} WHERE id = %s ORDER BY refresh_time DESC'''
...
            cursor.execute(sql, (id,))
            data = cursor.fetchall()
```

**Reply by XPPA0215** [reply]
Thanks, it's not a problem to handle in python.
However, this approach still carries the risk of SQL injection.
While in the majority of scenarios, the use of ID is reliable.

**Reply by methane** [reply, marked as answer]
`(id / 10) % 1000)` never have SQL injection risk.
Do not use untrusted input for table name, off course.

**Reply by XPPA0215** [reply]
I understand. Thank you for your reply
